src/actions/moon-phase.js

```diff
@@ -3,67 +3,27 @@ const Types = require('../action-types/moon-phase');
 
 const actions = exports;
 
-// Marks the moment we start waiting for the moon phase
-actions.beginLoad = () => {
-
-    return {
-        type: Types.MOON_LOAD_BEGIN
-    };
-};
-
-// Marks the moment we've received some good info about the moon phase
-actions.loaded = (moon) => {
-
-    return {
-        type: Types.MOON_LOADED,
-        payload: moon
-    };
-};
-
-// Marks the moment an error occurred while look-up the moon phase
-actions.loadError = (error) => {
-
-    return {
-        type: Types.MOON_LOAD_ERROR,
-        payload: error
-    };
-};
-
-// We'll use a thunk to orchestrate the actions (listed above)
-// of this asynchronous process of hitting an API
 actions.load = () => {
 
-    // We'll want to lookup the moon phase for this very millisecond
     const now = Date.now();
 
-    return (dispatch) => {
-
-        // Indicate to the app that we're beginning to load the moon phase
-        dispatch(actions.beginLoad());
-
-        // Make a request to the Farmsense API
-        const getMoonPhase = Axios.get(`http://api.farmsense.net/v1/moonphases/?d=${now}`);
+    return {
+        type: 'MOON_LOAD',
+        payload: new Promise((resolve, reject) => {
```

Comment:
Would not suggest creating your own promise here—it's clunky to use the `resolve()` and`reject()` callbacks, and they aren't necessary here.

```js
payload: Axios.get(`http://api.farmsense.net/v1/moonphases/?d=${now}`).then((response) => {

    // Format of response from Farmsense API looks like [{ ... }]
    const result = response.data[0];

    // Bad result
    if (!result || result.Error) {
        throw new Error(response);
    }

    // Looks good!
    return result;
})
```